test/e2e/sourcery.js

```diff
@@ -32,6 +33,32 @@ module.exports = {
         .assert.containsText(menuBar.activeNavigationTab, menuBar.timeLogging)
         .assert.cssProperty(menuBar.activeNavigationTab, 'color', common.blueColor )
         .assert.containsText(timeLogging.selectedDay, date.getDate())
+        
+    },
+
+    'Admin creates new task': function (browser) {
+        let taskName = 'nightwatch test ' + helper.randNumber();
+
+        browser
+        .url(browser.launchUrl + 'tasks')
+        .waitForElementVisible(common.pageTitle);
+
+        browser
+        .click(tasks.createTaskBtn)
+        .waitForElementVisible(common.userInfoTitle)
+        .setValue(tasks.nameField, taskName )
+        .setValue(tasks.descriptionField, 'nightwatch test desc')
+        .click(tasks.billDropdown)
+        .click(tasks.billYesOption)
+        .assert.containsText(tasks.billDropdown, 'Yes')
+        .setValue(tasks.rateField, '1') 
+        .click(tasks.saveTaskBtn)
+        .waitForElementVisible(common.userInfoTitle);
+            
+        browser.url(browser.launchUrl + 'tasks')
+        .waitForElementVisible(common.pageTitle)
+        .setValue(tasks.nameSearchField, taskName )
+        .waitForElementVisible(tasks.nameSearchResultField(taskName))
```

Comment:
Doesnt waitForElementVisible makes the assertion that in serach results there is field with created tasks name? Because in in console they count as passed assertions. 
![image](https://user-images.githubusercontent.com/42130184/44157583-778f2b00-a0bb-11e8-8814-67eba984562f.png)
 Both assert and WaitFor would stop further asserts if they would fail.

Reply:
_Selenium Assertions can be of three types: “assert”, “verify”, and ” waitFor”. When an “assert” fails, the test is aborted. When a “verify” fails, the test will continue execution, logging the failure. A “waitFor” command waits for some condition to become true._

Reply:
Sorry, my question wasn't clear.
1. Ideally for readability/clarity test should have some logic and clear assert which identifies what it tests/asserts. In your case there is quite a few things but yep, the test itself makes the thing it needs to do.
2. You tried to create something with name "some random string" but it failed and you got to search by "some random string" and there is no "some random string" but there is "some random string 123". Will the test fail or pass?

Reply:
The test would fail, because i search for a task with exact same name that i used in task creation. Thats why nameSearchResultField selector is generated using taskName.